Framework/CurveFitting/src/Functions/Meier.cpp

```diff
@@ -0,0 +1,272 @@
+// Mantid Repository : https: // github.com/mantidproject/mantid
+//
+// Copyright &copy; 2022 ISIS Rutherford Appleton Laboratory UKRI,
+// NScD Oak Ridge National Laboratory, European Spallation Source,
+// Institut Laue - Langevin &CSNS, Institute of High Energy Physics, CAS
+// SPDX - License - Identifier : GPL - 3.0 +
+#include "MantidCurveFitting/Functions/Meier.h"
+#include "MantidAPI/FunctionFactory.h"
+#include "MantidKernel/LambdaValidator.h"
+#include <cmath>
+#include <valarray>
+
+namespace Mantid::CurveFitting::Functions {
+
+using namespace CurveFitting;
+
+using namespace Kernel;
+
+using namespace API;
+
+DECLARE_FUNCTION(Meier)
+
+namespace {
+std::string isMultipleOf05(double value) {
+  double temp = value * 2;
+  return floor(temp) == ceil(temp) ? "" : "Spin value is not a multiple of 0.5";
+}
+
+double getSinSquared(const double &cosSquared) { return 1 - cosSquared; }
+
+double getCosSquared(const double &cos2squared) { return 0.5 * (1 + std::sqrt(cos2squared)); }
+} // namespace
+
+void Meier::init() {
+  declareParameter("A0", 0.5, "Amplitude");
+  declareParameter("FreqD", 0.01, "Angular Frequency due to dipolar coupling (MHz)");
+  declareParameter("FreqQ", 0.05,
+                   "Angular Frequency due to quadrupole interaction of the nuclear spin (MHz) due to a field gradient"
+                   "exerted by the presence of the muon");
+  declareParameter("Sigma", 0.2, "Gaussian decay rate");
+  declareParameter("Lambda", 0.1, "Exponential decay rate");
+  // J, Total angular momentum quanutm number
+  declareAttribute("Spin", API::IFunction::Attribute(3.5), Mantid::Kernel::LambdaValidator<double>(isMultipleOf05));
+}
+
+void Meier::function1D(double *out, const double *xValues, const size_t nData) const {
+  const double J = getAttribute("Spin").asDouble();
+
+  const double A0 = getParameter("A0");
+  const double J2 = round(2 * J);
+  const double Lambda = getParameter("Lambda");
+  const double Sigma = getParameter("Sigma");
+
+  std::valarray<double> xValArray(xValues, nData);
+
+  const std::valarray<double> gau = std::exp(-0.5 * pow((Sigma * xValArray), 2));
+  const std::valarray<double> Lor = std::exp(-Lambda * xValArray);
+
+  std::valarray<double> positiveLambda;
+  std::valarray<double> negativeLambda;
+  std::valarray<double> cos2AlphaSquared;
+
+  precomputeIntermediateSteps(cos2AlphaSquared, negativeLambda, positiveLambda, J2);
+
+  std::valarray<double> Px;
+  calculatePx(Px, xValArray, cos2AlphaSquared, negativeLambda, positiveLambda, J2);
+
+  std::valarray<double> Pz;
+  calculatePz(Pz, xValArray, cos2AlphaSquared, negativeLambda, positiveLambda, J2);
+
+  const std::valarray<double> outValArray = A0 * gau * Lor * (1. / 3.) * (2 * Px + Pz);
```

Comment:
looking at the doc, it is missing the`A0 * gau * Lor` factor. Could you please add it? Then the doc will match what the code is actually doing.